plenum/common/types.py

```diff
@@ -472,18 +457,30 @@ class ConsistencyProof(MessageBase):
     f.CONS_PROOF
 ])
 
-# class ConsProofRequest(MessageBase):
-#     typename = CONS_PROOF_REQUEST
-#     schema = (
-#         (f.LEDGER_ID.nm, LedgerIdField()),
-#         (f.SEQ_NO_START.nm, NonNegativeNumberField()),
-#         (f.SEQ_NO_END.nm, NonNegativeNumberField()),
-#     )
-ConsProofRequest = TaggedTuple(CONS_PROOF_REQUEST, [
-    f.LEDGER_ID,
-    f.SEQ_NO_START,
-    f.SEQ_NO_END
-])
+
+class MessageReq(MessageBase):
+    """
+    Purpose: ask node for any message
+    """
+    typename = MESSAGE_REQUEST
+    schema = (
+        (f.MSG_TYPE.nm, ChooseField(values={LEDGER_STATUS,
+                                            CONSISTENCY_PROOF, PREPREPARE})),
+        (f.PARAMS.nm, HetroMapField())
```

Comment:
It can be used to request arbitrary kinds of messages, and queries for different messages involves different params. The way you know is by looking at the handler